lib/App/Config/Chronicle.pm

```diff
@@ -347,6 +350,316 @@ sub current_revision {
     return $settings->{_rev};
 }
 
+######################################################
+###### New API stuff here
+######################################################
+
+has local_caching => (
+    isa     => 'Bool',
+    is      => 'ro',
+    default => 0,
+);
+
+=head2 update_cache
+
+Loads latest values from data chronicle into local cache.
+Calls to this method are rate-limited by <refresh_interval>.
+
+=cut
+
+sub update_cache {
+    my $self = shift;
+    die 'Local caching not enabled' unless $self->local_caching;
+
+    return unless $self->_has_refresh_interval_passed();
+    $self->_updated_at(time);
+
+    return unless $self->_is_cache_stale();
+
+    my $keys = [$self->_dynamic_keys(), '_global_rev'];
+    my @all_entries = $self->_retrieve_objects_from_chron($keys);
+    $self->_store_objects_in_cache({map { $keys->[$_] => $all_entries[$_] } (0 .. scalar @$keys - 1)});
+
+    return 1;
+}
+
+sub _has_refresh_interval_passed {
+    my $self                   = shift;
+    my $now                    = time;
+    my $prev_update            = $self->_updated_at;
+    my $time_since_prev_update = $now - $prev_update;
+    return ($time_since_prev_update > $self->refresh_interval);
+}
+
+sub _is_cache_stale {
+    my $self      = shift;
+    my @rev_cache = $self->_retrieve_objects_from_cache(['_global_rev']);
+    my @rev_chron = $self->_retrieve_objects_from_chron(['_global_rev']);
+    return !($rev_cache[0] && $rev_chron[0] && $rev_cache[0]->{data} eq $rev_chron[0]->{data});
+}
+
+=head2 global_revision
+
+Returns the global revision version of the config chronicle.
+This will correspond to the last time any of values were changed.
+
+=cut
+
+sub global_revision {
+    my $self = shift;
+    return ($self->get('_global_rev'));
+}
+
+=head2 set
+
+Takes a hashref of key->value pairs and atomically sets them in config chronicle
+
+Example:
+    set({key1 => 'value1', key2 => 'value2', key3 => 'value3',...});
+
+=cut
+
+sub set {
+    my ($self, $pairs) = @_;
+    my $rev_obj   = Date::Utility->new;
+    my $rev_epoch = $rev_obj->{epoch};
+
+    grep { die "Cannot set with key: $_ | Key must be defined with 'global: 1'" unless $self->_key_is_dynamic($_) } keys %$pairs;
```

Comment:
This void-context `grep` is better written as a `foreach` loop. Also consider putting the condition upfront

    $self->_key_is_dynamic($_) or die "Cannot set with key: $_ | Key must be defined with 'global: 1'"
        foreach keys %$pairs;